homeassistant/components/xiaomi_aqara.py

```diff
@@ -162,12 +184,43 @@ def stop_ringtone_service(call):
         else:
             _LOGGER.error('Unknown gateway sid: %s was specified.', gw_sid)
 
-    hass.services.async_register(DOMAIN, 'play_ringtone',
-                                 play_ringtone_service,
-                                 description=None, schema=None)
-    hass.services.async_register(DOMAIN, 'stop_ringtone',
-                                 stop_ringtone_service,
-                                 description=None, schema=None)
+    # pylint: disable=unused-variable
+    def add_device_service(call):
+        """Service to add a new sub-device within the next 30 seconds."""
+        gw_sid = call.data.get(ATTR_GW_MAC).replace(":", "").lower()
+        for (_, gateway) in hass.data[PY_XIAOMI_GATEWAY].gateways.items():
+            if gateway.sid == gw_sid:
+                join_permission = {'join_permission': 'yes'}
+                gateway.write_to_hub(gateway.sid, **join_permission)
+                hass.components.persistent_notification.async_create(
+                    'Join permission enabled for 30 seconds! '
+                    'Please press the pairing button of the new device once.',
+                    title='Xiaomi Aqara Gateway')
+                break
+        else:
+            _LOGGER.error('Unknown gateway sid: %s was specified.', gw_sid)
+
+    # pylint: disable=unused-variable
+    def remove_device_service(call):
+        """Service to remove a sub-device from the gateway."""
+        device_id = call.data.get(ATTR_DEVICE_ID)
+        gw_sid = call.data.get(ATTR_GW_MAC).replace(":", "").lower()
+        remove_device = {'remove_device': device_id}
+        for (_, gateway) in hass.data[PY_XIAOMI_GATEWAY].gateways.items():
+            if gateway.sid == gw_sid:
+                gateway.write_to_hub(gateway.sid, **remove_device)
+                break
+        else:
+            _LOGGER.error('Unknown gateway sid: %s was specified.', gw_sid)
+
+    for xiaomi_aqara_service in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[xiaomi_aqara_service].get(
+            'schema', XIAOMI_AQARA_SERVICE_SCHEMA)
+        service_handler = SERVICE_TO_METHOD[xiaomi_aqara_service].get('method')
```

Comment:
This is wrong and has broken the Xiaomi Aqara component. You are passing a string as service handler instead of the actual service. If you would have run this code locally, you would have noticed this right away 🤔 

There is no need to make it complicated and use SERVICE_TO_METHOD dict. Instead just directly register the services.